### DISCUSSION TITLE
Convert dispatch service into workflow

### DISCUSSION THREAD

**Discussion by bilalshaikh42** [biosimulations/biosimulations#4349] [Ideas]
The dispatch service has 2 main functions now.
- Monitor running jobs on the HPC
- Post-processing of simulation runs

The latter should be turned into a workflow using Argo Workflows. The dispatch service would generate the workflow definition from a template, and then submit it to the Argo workflows API (already installed). 

https://argoproj.github.io/argo-workflows/core-concepts/

**Comment by bilalshaikh42** [comment]
Actually, even the first function can be included as a part of the workflow.

**Comment by jonrkarr** [comment]
This seems like the right direction to go. This could also be used to support more complex runs, such as that use multiple tools e.g., for running different simulations and/or generating different outputs.
